samples/subsys/testsuite/pytest/shell/pytest/test_shell.py

```diff
@@ -2,54 +2,28 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-import time
 import logging
 
-from twister_harness import DeviceAdapter, TwisterHarnessTimeoutException
+import pytest
+from twister_harness import DeviceAdapter
+from twister_harness.helpers.shell import Shell
 
 logger = logging.getLogger(__name__)
 
 
-def wait_for_message(dut: DeviceAdapter, message: str, timeout: float = 20) -> bool:
-    time_started = time.time()
-    while True:
-        line = dut.readline(timeout=timeout)
-        if line:
-            logger.debug("#: " + line)
-        if message in line:
-            return True
-        if time.time() > time_started + timeout:
-            return False
-
-
-def wait_for_prompt(dut: DeviceAdapter, prompt: str = 'uart:~$', timeout: float = 20.0) -> bool:
-    """
-    Send every 0.5 second "enter" command to the device until shell prompt
-    statement will occur (return True) or timeout will be exceeded (return
-    False).
-    """
-    timeout_time = time.time() + timeout
-    while time.time() < timeout_time:
-        dut.write(b'\n')
-        try:
-            line = dut.readline(timeout=0.5)
-        except TwisterHarnessTimeoutException:
-            # ignore read timeout and try to send enter once again
-            continue
-        if prompt in line:
-            logger.debug('Got prompt')
-            dut.clear_buffer()
-            return True
-    return False
-
-
-def test_shell_print_help(dut: DeviceAdapter):
-    assert wait_for_prompt(dut)
-    dut.write(b'help\n')
-    assert wait_for_message(dut, "Available commands")
-
-
-def test_shell_print_version(dut: DeviceAdapter):
-    assert wait_for_prompt(dut)
-    dut.write(b'kernel version\n')
-    assert wait_for_message(dut, "Zephyr version")
+@pytest.fixture(scope='function')
+def shell(dut: DeviceAdapter) -> Shell:
+    """Return ready to use shell interface"""
+    shell = Shell(dut)
+    assert shell.wait_for_prompt()
+    return shell
+
+
+def test_shell_print_help(shell: Shell):
+    lines = shell.write_and_get_response('help')
```

Comment:
Można by się nad nazwą zastanowić. Write to bardziej pisanie bezpośrednio do bufora ... tutaj już mamy shell'a, więc moze coś w stylu
- np. exec_command (np. https://docs.paramiko.org/en/2.4/api/client.html)
- albo może coś zapożyczyć z suprocess: run, call, get_output itp.